app/controllers/planning_applications_controller.rb

```diff
@@ -250,4 +267,8 @@ def validation_notice_mail
   def ensure_user_is_reviewer
     render plain: "forbidden", status: 403 and return unless current_user.reviewer?
   end
+
+  def ensure_constraint_edits_unlocked
```

Comment:
Would it not make more sense to have a "constraints" controller?

Reply:
I'm thinking it'd be a nice way to separate the code, having a constraints controller with "create" and "new" actions for the constraints and dedicated helper methods / model validations for later 🤔